jerry-core/api/jerry.c

```diff
@@ -5513,6 +5514,185 @@ jerry_get_user_value (const jerry_value_t value) /**< jerry api value */
   return ecma_copy_value (CBC_SCRIPT_GET_USER_VALUE (script_p));
 } /* jerry_get_user_value */
 
+/**
+ * Returns a newly created source info structure corresponding to the passed script/module/function.
+ *
+ * @return a newly created source info, if at least one field is available, NULL otherwise
+ */
+jerry_source_info_t *
+jerry_get_source_info (const jerry_value_t value) /**< jerry api value */
+{
+  jerry_assert_api_available ();
+
+#if JERRY_FUNCTION_TO_STRING
+  jerry_source_info_t source_info;
+
+  source_info.enabled_fields = 0;
+  source_info.source_code = ECMA_VALUE_UNDEFINED;
+  source_info.function_arguments = ECMA_VALUE_UNDEFINED;
+  source_info.source_range_start = 0;
+  source_info.source_range_length = 0;
+
+  if (!ecma_is_value_object (value))
```

Comment:
If we do a return here, do we even need to initialize the source_info above? Maybe we could do it after this check?